src/objects/zcl_abapgit_object_suso.clas.abap

```diff
@@ -136,25 +156,92 @@ CLASS zcl_abapgit_object_suso IMPLEMENTATION.
 
   METHOD zif_abapgit_object~delete.
 
-    DATA: lv_object TYPE tobj-objct.
+    " FM SUSR_DELETE_OBJECT calls the UI. Therefore we reimplement it here.
+
+    CONSTANTS:
+      co_act_delete TYPE activ_auth VALUE '06'.
+
+    DATA:
+      lv_act_head TYPE cl_suso_gen=>td_act,
+      lv_dummy    TYPE sy-langu,
+      lo_suso     TYPE REF TO cl_suso_gen,
+      lv_docu_obj TYPE dokhl-object,
+      lv_failed   TYPE abap_bool.
+
+    CREATE OBJECT lo_suso.
+
+    lv_failed = lo_suso->suso_load_from_db( mv_objectname ).
+
+    IF lv_failed = abap_true.
+      " Object & does not exist; choose an existing object
+      MESSAGE s111(01) WITH mv_objectname INTO lv_dummy.
+      zcx_abapgit_exception=>raise_t100( ).
+    ENDIF.
 
+    "initialize dialog control data
+    lo_suso->get_suso_edit_mode( EXPORTING id_object      = mv_objectname
+                                           id_planed_act  = co_act_delete
+                                 IMPORTING ed_mode_head   = lv_act_head ).
+    IF lv_act_head <> co_act_delete.
+      zcx_abapgit_exception=>raise( |AUTH { mv_objectname }: Delete not allowed| ).
+    ENDIF.
 
-    lv_object = ms_item-obj_name.
-    CALL FUNCTION 'SUSR_DELETE_OBJECT'
+    IF lo_suso->suso_collect_in_cts( mv_objectname ) IS NOT INITIAL.
+      RETURN.
+    ENDIF.
+
+    "delete documentation in all languages
+    lv_docu_obj  = mv_objectname.
```

Comment:
suggest moving the deletion of documentation to separate method, this will also avoid having comments in the code 😄

Reply:
fixed